x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/structurefinder/TextStructureUtils.java

```diff
@@ -417,18 +500,47 @@ static Map<String, String> guessScalarMapping(
         Collection<String> fieldValues,
         TimeoutChecker timeoutChecker,
         boolean ecsCompatibility
+    ) {
+        return guessScalarMapping(explanation, fieldName, fieldValues, timeoutChecker, ecsCompatibility, null);
+    }
+
+    /**
+     * Given some sample values for a field, guess the most appropriate index mapping for the
+     * field.
+     * @param explanation List of reasons for choosing the overall text structure.  This list
+     *                    may be non-empty when the method is called, and this method may
+     *                    append to it.
+     * @param fieldName Name of the field for which mappings are to be guessed.
+     * @param fieldValues Values of the field for which mappings are to be guessed.  The guessed
+     *                    mapping will be compatible with all the provided values.  Must not be
+     *                    empty.
+     * @param timeoutChecker Will abort the operation if its timeout is exceeded.
+     * @param ecsCompatibility The mode of compatibility with ECS Grok patterns.
+     * @param timestampFormatOverride The format of the timestamp as given in the request overrides.
+     * @return The sub-section of the index mappings most appropriate for the field,
+     *         for example <code>{ "type" : "keyword" }</code>.
+     */
+    static Map<String, String> guessScalarMapping(
+        List<String> explanation,
+        String fieldName,
+        Collection<String> fieldValues,
+        TimeoutChecker timeoutChecker,
+        boolean ecsCompatibility,
+        String timestampFormatOverride
     ) {
         assert fieldValues.isEmpty() == false;
 
         if (fieldValues.stream().allMatch(value -> "true".equals(value) || "false".equals(value))) {
             return Collections.singletonMap(MAPPING_TYPE_SETTING, "boolean");
         }
 
-        try {
-            return findTimestampMapping(explanation, fieldValues, timeoutChecker, ecsCompatibility);
-        } catch (IllegalArgumentException e) {
-            // To be mapped as type "date" all the values must match the same timestamp format - if
-            // they don't we'll end up here, and move on to try other possible mappings
+        if (NULL_TIMESTAMP_FORMAT.equals(timestampFormatOverride) == false) {
```

Comment:
Maybe we shouldn't do this check.  `timestamp_format` applies to the _primary_ timestamp that we'll put into `@timestamp`.  But delimited files can contain multiple timestamp columns and we used to try to identify all of them.  So I think that even if the user says there's no timestamp format we should still try to do this, just not call any of them `@timestamp`.  (This issue is also of secondary importance, because the primary case where a timestamp is required is in the semi-structured log files where previously we'd refuse to analyze the file if it had no timestamp, but this code is for fully structured formats where they're already split into fields.)